### PR TITLE
Generate project website

### PR DESCRIPTION
**Pull request by russgold**
After release, this will be published to Github. In the meantime, after you build, do:

mvn site site:stage
cd target/staging
python -m SimpleHTTPServer 9999

Then browse to localhost:9999

You will note menu with links to a few of the pages (as a demonstration; I wasn't sure which should be included)

Click the "domainResource" link to browse the domain model. Then click on DomainSpec to expand it.

We can probably improve on the descriptions.

### CODE REVIEW

**Comment by markxnelson** [comment]
can you please help me understand why this change is being introduced? what was wrong with the existing documentation using markdown and github pages?

File: src/site/xhtml/domain-resource.xhtml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <title>WebLogic Kubernetes Operator:Custom Domain Resource</title>
+</head>
+<body>
+    <h2>Custom Domain Resource</h2>
+
+The WebLogic Kubernetes Operator requires definition of.
+
+
+
+Blah blah need some words here.
```
[the remark this answers is not in the document]
Comment (russgold) [reply src/site/xhtml/domain-resource.xhtml]: Yes, I am not sure what is needed there; I can certain make something up :)

**Comment by russgold** [comment]
Generating the website has a number of advantages over just using markdown, including:

- the ability to include generated content. This PR adds documentation for the domain resource, which is generated directly from the code, and therefore is guaranteed to match the implementation. I am also working on generating documentation for the REST api.
- the ability to surface information about the project, including the team and actual distribution information, as recorded in the master pom and used during distribution
- auto-generation of a menu on each (selected) page, to ease navigation of the site.

The site is not generated by default, but only when the `mvn site` command is given. When we are ready to publish it, we issue `mvn site-deploy`

**Comment by rjeberhard** [comment]
So, @russgold...  What I was expecting is that the markdown section would stay as *.md files and then the generated content would continue to be under /site and be linked.

I don't really get why we would want to pre-generate the *.md files.

**Comment by russgold** [comment]
The .md files are as they are, but html is generated from them. That makes the site a coherent whole, including providing for the menu. Have you tried viewing the site as described in the description?

If you don't want the navigation, I'll have to find a new way to introduce the generated content. GitHub pages are taken directly from the git repository, while the build generates content to target directories. I'd also have to do some research to figure out if I can embed Docson (the schema navigation tool) into a markdown page.

Is there some disadvantage you see in generating the site rather than manually editing it as markdown?

**Comment by rjeberhard** [comment]
You're going to have to show us... There are two sites...  The .md files generate https://github.com/oracle/weblogic-kubernetes-operator while /site is the content for https://oracle.github.io/weblogic-kubernetes-operator/

**Comment by rjeberhard** [comment]
So, I was expecting the generated content to be under /site and to have the .md files link there for that generated content

**Comment by rjeberhard** [comment]
Let's defer this for 2.0.

**Pull request closed without merging**